### PR TITLE
Add segmentId to external links

### PR DESCRIPTION
**Pull request by ahmadmoawad**
- Add source specific segment ids to `betterPromo` & `magnetslot`
- Add a segmentId query string for all external event/course/forum promo links in the magnet slot

### CODE REVIEW

**Review by maxcbc — CHANGES_REQUESTED** [review]
We'll also need to do the same for forumPromos: https://github.com/Financial-Times/n-magnet/blob/main/src/components/forumpromo/main.js

File: src/components/forumpromo/main.js [hunk 1]
```diff
@@ -2,17 +2,20 @@
...
 import {h, render} from '@financial-times/x-engine';
 import {dispatchTrackingEvent} from '../../lib/tracking';
+import {getMappedData} from './eventpromo-utils';
 
-export function renderForumpromo(magnetPlaceholderSelector, data) {
+export function renderForumpromo (magnetPlaceholderSelector, data) {
 	try {
-		const promoElement = <Forumpromo {...data} />;
+		const formattedData = getMappedData(data);
+		const promoElement = <Forumpromo {...formattedData} />;
```
Comment (maxcbc) [inline src/components/forumpromo/main.js:10]: i don't think that the data for a forumpromo is the same shape as that of an eventpromo, so I don't think using the same `getMappedData()` function will work.

A forumpromo's data has this format: https://github.com/Financial-Times/n-magnet/blob/main/demos/fixtures/promos/forumpromo.json
Whereas an eventpromo's data looks like this: https://github.com/Financial-Times/n-magnet/blob/main/demos/fixtures/promos/eventpromo/ft-live.json

Comment (maxcbc) [reply src/components/forumpromo/main.js:10]: Have you managed to get the demo running? It might help you understand what data gets used where.

**Review by maxcbc — CHANGES_REQUESTED** [review]

File: src/components/forumpromo/main.js (hunk 1 above)
Comment (ahmadmoawad) [reply src/components/forumpromo/main.js:10]: @maxcbc Yes, I have run the demo. It's not clear not to update which part in the https://github.com/Financial-Times/n-magnet/blob/main/src/components/forumpromo/main.js from the previous comment.

File: src/components/forumpromo/main.js [hunk 2]
```diff
@@ -3,16 +3,21 @@ import {Forumpromo} from '@financial-times/n-eventpromo';
 import {h, render} from '@financial-times/x-engine';
 import {dispatchTrackingEvent} from '../../lib/tracking';
 
-export function renderForumpromo(magnetPlaceholderSelector, data) {
+const segmentId = 'a3ced873-2d86-959c-46de-b9eab4c8f8d0';
+
+export function renderForumpromo (magnetPlaceholderSelector, data) {
 	try {
+		const forumUrl = new URL(data.link);
+		forumUrl.searchParams.set('segmentId', segmentId);
```
Comment (maxcbc) [inline src/components/forumpromo/main.js:11]: I think you forgot to set `data.link` to the new url with `data.link = forumUrl.href`

Comment (maxcbc) [reply src/components/forumpromo/main.js:11]: Other than that this looks god

**Review by maxcbc — CHANGES_REQUESTED** [review]

**Review by maxcbc — APPROVED** [review]

**Review by AliceArmstrong — APPROVED** [review]
👍

**Pull request merged**